### ISSUE TITLE
`List error -402` when trying to sync (with gpodder)

### ISSUE THREAD

**Issue by alexanderadam** [lucasasselli/garmin-podcasts#9]
Somehow I still cannot sync.
I'm still trying to use [gpodder](https://gpodder.net/) but whenever I want to sync I'm getting `List error -402`.

Any advice?

**Comment by lucasasselli** [comment]
Hi Alexander,

This is (unfortunately) a memory issue. How many podcasts are you subscribed to on gPodder?

**Comment by alexanderadam** [comment]
Hi Luca,

> This is (unfortunately) a memory issue.

Would it be possible to change the error message in that case? Something like `Unfortunately you subscribed to too many feeds. Try to decrease the amount of feeds and sync again`.

> How many podcasts are you subscribed to on gPodder?

It depends. gPodder has feeds per device but also global feeds.

I thought I should be subscribed to some on the watch as well but gpodder claims that I `don't have any podcasts subscribed on this device`.

<div align="center">

![gPodder error: You don't have any podcasts subscribed on this device. ](https://user-images.githubusercontent.com/372620/113438064-1b9e7f80-93e8-11eb-8ab2-a1a82c702be5.png)

</div>

**Comment by lucasasselli** [comment]
Hi Alexander,

Thank you for going the extra mile here with the gPodder response! 

The most recent version of the app doesn't use the device ID anymore, and just download all the subscriptions. I can see that it was a short sighted decision on my part. I am restoring the old behavior, making the device ID optional.

**Comment by lucasasselli** [comment]
Version 3.0.2 restored the device ID field and has more descriptive error message. Let me know if it works!

**Comment by alexanderadam** [comment]
Thank you Luca! Now I'm not getting `List error -402` anymore but the `Sync` button seem to be non-functional now. It's becoming grey (probably the selected state) but nothing else happens.

Should I open another issue for this?

**Comment by lucasasselli** [comment]
Can you try to uninstall and re-install the app?

**Comment by lucasasselli** [comment]
Let's keep this one open, I was just able to recreate the issue. I'll upload a fix tonight or tomorrow!

**Comment by lucasasselli** [comment]
Fixed in version 3.0.3.

**Comment by alexanderadam** [comment]
> Fixed in version 3.0.3.

My phone says that I have now have `3.0.3` installed but the `Sync` button still doesn't work for me. 🤷

**Comment by lucasasselli** [comment]
Can you send me the crash dump? 

If you connect your watch to a PC/Mac under /GARMIN/APPS/LOGS you will find a file called CIQ_LOG.yml.

**Comment by alexanderadam** [comment]
I can but I'm unsure whether this will help you:

```yaml
---
Error: Unhandled Exception
Time: 2021-04-07T16:36:13Z
Part-Number: 006-B3113-00
Firmware-Version: '6.20'
Language-Code: eng
ConnectIQ-Version: 3.2.5
Store-Id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Store-Version: 15
Filename: AB3L3118
Appname: Podcasts
Stack: 
  - pc: 0x10004250
  - pc: 0x10000de1
  - pc: 0x100004fb
  - pc: 0x10002f17
  - pc: 0x10002dd0
```

**Comment by lucasasselli** [comment]
Just uploaded 3.0.4, hopefully this should be okay! Keeping my fingers crossed! If the issue is still there let me know!

**Comment by alexanderadam** [comment]
I tried it and it is in the `Loading` spinner screen since roughly an hour.
Is there any reliable way to see whether it is actually working or whether it is stuck somewhere?

**Comment by lucasasselli** [comment]
I think the timeout is something like 3 minutes. I don't see this issue on my side. Probably I can try an debug it with another crash dump. Sorry for the trouble. 😢

**Comment by alexanderadam** [comment]
> I think the timeout is something like 3 minutes.

I stopped it after two hours.

> Probably I can try an debug it with another crash dump.

```yaml

---
Error: Unhandled Exception
Time: 2021-04-08T16:14:18Z
Part-Number: 006-B3113-00
Firmware-Version: '6.20'
Language-Code: eng
ConnectIQ-Version: 3.2.5
Store-Id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Store-Version: 16
Filename: AB3L3118
Appname: Podcasts
Stack: 
  - pc: 0x100010a3
```

So are you saying that this stack trace is useful for you? 😮

> Sorry for the trouble. :cry:

No worries. IMHO you are are helpful and very responsive maintainer. :+1:

**Comment by lucasasselli** [comment]
Hi Alexander, the issue should be fixed in version 3.0.7! I was able to reproduce the bug on my device. Please let me know if you see any other issue!

**Comment by alexanderadam** [comment]
Luca, you're awesome!
Now it's working! Thank you so much!

**Comment by alexanderadam** [comment]
Sorry. I was wrong. After selecting the button I can select some episodes. When I go back, I'm getting the message whether I want to sync.
If I confirm, I'm getting the endless sync dialogue again.

**Comment by lucasasselli** [comment]
Hi Alexander, other users are confirming that 3.1.0 fixed all issues for manual sync. Can you confirm it?

**Comment by alexanderadam** [comment]
Unfortunately not.

![Error! -1005, -1005, -1005, -1005](https://user-images.githubusercontent.com/372620/114402032-c01c8080-9ba3-11eb-9978-77ed1ee1f76f.jpg)

**PS:** There's no `CIQ_LOG.yml`, only this `CIQ_LOG.BAK` (probably from version before):

```yaml

---
Error: Symbol Not Found Error
Details: "Could not find symbol '008000a7'"
Time: 2021-04-11T21:30:00Z
Part-Number: 006-B3113-00
Firmware-Version: '6.20'
Language-Code: eng
ConnectIQ-Version: 3.2.5
Store-Id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Store-Version: 73
Filename: ABA80808
Appname: GearMin 2.0
Stack: 
  - pc: 0x100025d1
  - pc: 0x10002403
  - pc: 0x10000e8e
```

**Comment by lucasasselli** [comment]
Hi Alexander, the dump you sent is not from Podcast, but from another app. Regardless, the error you see is due to an unexpected format from the podcast provider. What podcast are you downloading?

**Comment by alexanderadam** [comment]
> Hi Alexander, the dump you sent is not from Podcast, but from another app.

Yes, this is what I wrote in the sentence just before the dump. 😉 

> Regardless, the error you see is due to an unexpected format from the podcast provider.

In this case a better error message for this would be great, too

> What podcast are you downloading?

It doesn't seem to matter in my case. I tried downloading episodes from [Nusschale](https://www.nussschale-podcast.de/feed/m4a/), [requestforcomments](https://requestforcomments.de/feed/m4a) and [minkorrekt](https://minkorrekt.de/feeds/). Feed data is still coming from gpodder.
I'm selecting `Sync`, then some episodes. Afterwards I go back and let it download. The status circle grows until it should reach 100% and then this error appears.

So far I wasn't able to find any working episode.
